### PR TITLE
RideShare

### PR DESCRIPTION
**Pull request by amanzan1**
# Ride Share
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe a design decision you had to make when working on this project. What options were you considering? What helped you make your final decision? | I think the design decision I was considering and the thing I was having most trouble was determining in my mind how these were all interrelated.
Describe any examples of composition that you encountered in this project, if any | Well everything was interrelated for example the trip dispatcher had their own methods , and the tripdispatch,passenger,trip,and driver were all interacting to make the trip possible.
Describe a _nominal test_ that you wrote for this assignment. |
Describe an _edge case test_ that you wrote for this assignment. |
Describe a concept that you gained more clarity on as you worked on this assignment. | I was becoming a little more comfortable with testing.

### CODE REVIEW

File: lib/passenger.rb [hunk 1]
```diff
@@ -17,8 +17,45 @@ def get_drivers
...
+     @trips.each do |trip|
+       total += trip.cost
+     end
+     return total
+    end
+
+    def total_time
+      total = 0
+      @trips.each do |trip|
+        total = trip.end_time - trip.start_time
```
Comment (CheezItMan) [inline lib/passenger.rb:31]: You should use the `trip_duration` method you just wrote for this.

File: lib/passenger.rb [hunk 2]
```diff
@@ -17,8 +17,45 @@ def get_drivers
...
+
+
     def add_trip(trip)
       @trips << trip
     end
   end
 end
+
+
+ #pseudocode - describe "get total_rev do
```
Comment (CheezItMan) [inline lib/passenger.rb:44]: It's good that you're putting pseudocode here to describe what you're trying to do.

File: lib/trip_dispatcher.rb [hunk 3]
```diff
@@ -61,8 +69,40 @@ def load_passengers
...
+        id: @trips.length + 1,
+        driver: driver,
+        passenger: passenger,
+        start_time: Time.now, #current-time- what is current-time?
+        end_time: nil,
+        cost: nil,
+        rating: nil
+      }
+
+        trip = Trip.new(requested_trip)
```
Comment (CheezItMan) [inline lib/trip_dispatcher.rb:98]: You are not adding the trip to the passenger or driver's attributes.

File: specs/driver_spec.rb [hunk 4]
```diff
@@ -55,10 +55,27 @@
...
 
+  describe "driver total revenue" do
+    before do
+      pass = RideShare::Passenger.new(id: 1, name: "Ada", phone: "412-432-7640")
+      @driver = RideShare::Driver.new(id: 3, name: "Lovelace", vin: "12345678912345678")
+      @trip = RideShare::Trip.new({id: 8, driver: @driver, passenger: pass, start_time: Time.new("2016-08-08"),cost:3.50, end_time: Time.new("2016-08-09"), rating: 5})
+      @driver.add_trip(@trip)
+    end
+
+    it "calculates driver total revenue" do
```
Comment (CheezItMan) [inline specs/driver_spec.rb:66]: You should also have a test for what the total revenue is when there are no trips, and with several trips.

File: lib/driver.rb [hunk 5]
```diff
@@ -27,20 +27,29 @@ def average_rating
...
     end
 
+    def total_revenue
+      total_revenue = 0
+      @trips.each do |trip|
+        total_revenue += (trip.cost - 1.65) * 0.80
+      end
+
+      return total_revenue.round(2)
+    end
```
Comment (CheezItMan) [inline lib/driver.rb:46]: What about the average revenue per hour method?

**Comment by CheezItMan** [comment]
# Ride Share
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Used Git Regularly | You do not have many commit messages.  Also your commit messages focus on waves, etc and not on functionality added.  
Answer comprehension questions | Your difficulty with seeing how things interrelate is **not** a design decision.  You also do not describe composition.  You also have 2 questions **unanswered**.  
**Wave 1** |
`Trip` has a helper method to calculate duration | Yes, but it doesn't check to see if `@end_time` and `@start_time` are not nil.
`Passenger` has a method to calculate total cost of all trips | Check
`Driver` has a method to calculate total revenue |  Check, but see my notes on the missing average revenue per hour method.
Complete tests |  You have **some** tests.  See my notes on missing edge cases.  
**Wave 2** |
`TripDispatcher` has a new method to create trips | Check, with some bugs.
Creating a trip in `TripDispatcher` relies on methods in `Driver` and `Passenger` to modify their own attributes | Check
Methods from wave 1 work on `Driver`s and `Passenger`s with incomplete trips |  **NOPE**
Complete tests | Some missing edge cases.  **NO TESTS FOR TripDispatcher**
**Overall** | Let me be clear, there's a lot of improvement needed here.  Given the amount of support and tutoring, I would expect much more done at least having basic testing in each class and the methods completed.  You need to work on your understanding of how composition works, and writing effective tests.  When we meet tomorrow lets review this, and then look more closely at Hotel.
